### PR TITLE
net: lib: nrf_cloud: use log level Kconfig template

### PR DESCRIPTION
**Pull request by lemrey**
Use Kconfig template to define nRF Cloud log level.

### CODE REVIEW

File: subsys/net/lib/nrf_cloud/Kconfig [hunk 1]
```diff
@@ -44,4 +36,10 @@ config NRF_CLOUD_PORT
 config NRF_CLOUD_IPV6
 	bool "Configure nRF Cloud library to use IPv6 addressing. Otherwise IPv4 is used."
 
+module=NRF_CLOUD
+module-dep=LOG
```
Comment (lemrey) [inline subsys/net/lib/nrf_cloud/Kconfig]: Should we depend on `LOG` or `NET_LOG` @rlubos ?
I am using the networking subsystem Kconfig template.

Comment (rlubos) [reply subsys/net/lib/nrf_cloud/Kconfig]: I'd say to stick to the `LOG` - less modules inbetween means less problems to inherit from upstream. Rember the MQTT, right?

File: subsys/net/lib/nrf_cloud/Kconfig [hunk 2]
```diff
@@ -44,4 +36,10 @@ config NRF_CLOUD_PORT
...
+module-str=Log level for nRF Cloud
+module-help=Enables nRF Cloud log messages.
+source "${ZEPHYR_BASE}/subsys/net/Kconfig.template.log_config.net"
```
Comment (rlubos) [inline subsys/net/lib/nrf_cloud/Kconfig]: There's a regular `LOG` template as well, would it be possible to use it instead?
`subsys/logging/Kconfig.template.log_config`

Comment (lemrey) [reply subsys/net/lib/nrf_cloud/Kconfig]: Yeah, better do that.

**Pull request merged**